### PR TITLE
NEXT-5387 | Image gallery only supports 'cover' as displayMode.

### PR DESCRIPTION
**Pull request by sahinvardar**
Add support for displayMode : 'contain' for image gallary.

 * added is-contain check for gallery-slider-single-image css element.
 * added is-contain check in cms-element-image-gallery.html.twig file.

### 1. Why is this change necessary?

It would good to have the option to use displayMode: 'contain', which would make sure that the whole image would fit into the given image box in the image gallery.

### 2. What does this change do, exactly?

It add support for displayMode: 'contain' for images in the image gallery.

### 4. Please link to the relevant issues (if any).

[NEXT-5387 | Image gallery only supports 'cover' as displayMode](https://issues.shopware.com/issues/NEXT-5387)

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
        I did a manual test by changing the displayMode to 'contain'.
- [x] I have squashed any insignificant commits
- [x] I have written or adjusted the documentation according to my changes
        Not sure if this is needed by this change.
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Storefront/Resources/views/page/product-detail/index.html.twig [hunk 1]
```diff
@@ -29,7 +29,7 @@
                                                 'mediaItems': mediaItems,
                                                 'zoom': true,
                                                 'zoomModal': true,
-                                                'displayMode': 'cover',
+                                                'displayMode': 'contain',
```
Comment (keulinho) [inline src/Storefront/Resources/views/page/product-detail/index.html.twig]: Why did you change the display mode here?

**Comment by keulinho** [comment]
Thx for your Contribution @SakirSoft :tada: ,

i have just a little question

File: src/Storefront/Resources/views/page/product-detail/index.html.twig (hunk 1 above)
Comment (sahinvardar) [reply src/Storefront/Resources/views/page/product-detail/index.html.twig]: I would say my personal opinion got in my way here :). From my point of view I can't understand how not to use 'contain' as default :), but yeah I can revert that back to 'cover', I also notice this is default in other places in ShopWare as well. 

Do you want me to revert that back to 'cover' ?

Btw begin on this matter, can you tell me or point me out on how to change the display mode for the image gallery through the ShopWare admin UI? Especially for a ProductListining page? 

I see code which would get the display mode through the corresponding twig file. But I wonder how this is expose in the UI.

Comment (keulinho) [reply src/Storefront/Resources/views/page/product-detail/index.html.twig]: Yes please revert this change back to cover ;)

You can change the display mode in the settings of the gallery cms block:
image

Comment (sahinvardar) [reply src/Storefront/Resources/views/page/product-detail/index.html.twig]: Updated the commit and reverted contain to cover back.

**Comment by sahinvardar** [comment]
Updated commit message to reflect revering back to displayMode cover as default.

**Comment by pantrtxp** [comment]
Hi @SakirSoft I was just about to import ist PR and noticed that you have not added the select option for the contain setting in the administration. So it would not be possible so set the gallery to 'contain' in the page builder.

Could you please add the Administration part including CSS to display the contain option accordingly

**Comment by sahinvardar** [comment]
Hi @pantrtxp ,

sure I can update the admin part as well. BTW let me redo the commit and get back to you.
Sadly I did commit on master at my fork. I will create a dedicated branch on my side because I have also another fix I want to contribute.


Best
SakirSoft

**Comment by sahinvardar** [comment]
Closing this in favour of [this](https://github.com/shopware/platform/pull/313) pull request.

**Comment by sahinvardar** [comment]
@pantrtxp 
Can you please check out the [new](https://github.com/shopware/platform/pull/313) pull request.
This also solves the  administration part.
